cluster-autoscaler/cloudprovider/konvoy/konvoy_manager.go

```diff
@@ -97,61 +118,31 @@ func (k *KonvoyManager) GetNodeGroupSize(nodeGroup string) (int, error) {
 	return len(nodes.Items), nil
 }
 
-// GetNodeGroupTargetSize returns the size of the node group as a sum of current
-// observed size and number of upcoming nodes.
-func (k *KonvoyManager) GetNodeGroupTargetSize(nodeGroup string) (int, error) {
-	k.nodeGroupQueueSizeLock.Lock()
-	defer k.nodeGroupQueueSizeLock.Unlock()
-	realSize, err := k.GetNodeGroupSize(nodeGroup)
-	if err != nil {
-		return realSize, err
+// GetNodeGroupTargetSize returns the target size of the node group.
+func (k *KonvoyManager) GetNodeGroupTargetSize(nodeGroupName string) (int, error) {
+	konvoyCluster := &kommanderv1beta1.KonvoyCluster{}
+	konvoyCluster.Name = k.clusterName
+	clusterNamespacedName := types.NamespacedName{
+		Namespace: "kommander",
+		Name:      konvoyCluster.Name,
 	}
-	return realSize + k.nodeGroupQueueSize[nodeGroup], nil
-}
-
-// SetNodeGroupSize changes the size of node group by adding or removing nodes.
-func (k *KonvoyManager) SetNodeGroupSize(nodeGroup string, size int) error {
-	currSize, err := k.GetNodeGroupTargetSize(nodeGroup)
-	if err != nil {
-		return err
+	if err := k.dynamicClient.Get(context.Background(), clusterNamespacedName, konvoyCluster); err != nil {
+		return unknownTargetSize, err
 	}
-	switch delta := size - currSize; {
-	case delta < 0:
-		absDelta := -delta
-		nodes, err := k.GetNodeNamesForNodeGroup(nodeGroup)
-		if err != nil {
-			return err
-		}
-		if len(nodes) < absDelta {
-			return fmt.Errorf("can't remove %d nodes from %s nodegroup, not enough nodes: %d", absDelta, nodeGroup, len(nodes))
-		}
-		for i, node := range nodes {
-			if i == absDelta {
-				return nil
-			}
-			if err := k.RemoveNodeFromNodeGroup(nodeGroup, node); err != nil {
-				return err
-			}
-		}
-	case delta > 0:
-		k.nodeGroupQueueSizeLock.Lock()
-		defer k.nodeGroupQueueSizeLock.Unlock()
-		for i := 0; i < delta; i++ {
-			k.nodeGroupQueueSize[nodeGroup]++
-			if err := k.addNodeToNodeGroup(nodeGroup); err != nil {
-				k.nodeGroupQueueSize[nodeGroup]--
-				return err
-			}
+
+	for _, nodePool := range konvoyCluster.Spec.ProvisionerConfiguration.NodePools {
+		if nodePool.Name == nodeGroupName {
+			return int(nodePool.Count), nil
```

Comment:
GetNodeGroupTargetSize == nodePool.Count. I don't understand it. This should be the max value of the autoscaling pool.

Reply:
Max value is user provided max configuration that sets limits for autoscaler. The Count is what autoscaler has decided on.

Reply:
👍